docs/docs/use_cases/extraction/how_to/examples.ipynb

```diff
@@ -0,0 +1,447 @@
+    "    tool_calls: List[BaseModel]  # Instances of pydantic model that should be extracted\n",
+    "\n",
+    "\n",
+    "def tool_example_to_messages(example: Example) -> List[BaseMessage]:\n",
+    "    \"\"\"Convert an example into a list of messages that can be fed into an LLM.\n",
+    "\n",
+    "    This code is an adapter that converts our example to a list of messages\n",
+    "    that can be fed into a chat model.\n",
+    "\n",
+    "    The list of messages per example corresponds to:\n",
+    "\n",
+    "    1) HumanMessage: contains the content from which content should be extracted.\n",
+    "    2) AIMessage: contains the extracted information from the model\n",
+    "    3) ToolMessage: contains confirmation to the model that the model requested a tool correctly.\n",
+    "\n",
+    "    The ToolMessage is required because some of the chat models are hyper-optimized for agents\n",
+    "    rather than for an extraction use case.\n",
+    "    \"\"\"\n",
+    "    messages: List[BaseMessage] = [HumanMessage(content=example[\"input\"])]\n",
+    "    openai_tool_calls = []\n",
+    "    for tool_call in example[\"tool_calls\"]:\n",
+    "        openai_tool_calls.append(\n",
+    "            {\n",
+    "                \"id\": str(uuid.uuid4()),\n",
+    "                \"type\": \"function\",\n",
+    "                \"function\": {\n",
+    "                    # The name of the function right now corresponds\n",
+    "                    # to the name of the pydantic model\n",
+    "                    # This is implicit in the API right now,\n",
+    "                    # and will be improved over time.\n",
+    "                    \"name\": tool_call.__class__.__name__,\n",
+    "                    \"arguments\": tool_call.json(),\n",
+    "                },\n",
+    "            }\n",
+    "        )\n",
+    "    messages.append(\n",
+    "        AIMessage(content=\"\", additional_kwargs={\"tool_calls\": openai_tool_calls})\n",
+    "    )\n",
+    "    tool_outputs = example.get(\"tool_outputs\") or [\n",
+    "        \"You have correctly called this tool.\"\n",
+    "    ] * len(openai_tool_calls)\n",
+    "    for output, tool_call in zip(tool_outputs, openai_tool_calls):\n",
+    "        messages.append(ToolMessage(content=output, tool_call_id=tool_call[\"id\"]))\n",
+    "    return messages"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "463aa282-51c4-42bf-9463-6ca3b2c08de6",
+   "metadata": {},
+   "source": [
+    "Next let's define our examples and then convert them into message format."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "7f59a745-5c81-4011-a4c5-a33ec1eca7ef",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "examples = [\n",
+    "    (\n",
+    "        \"The ocean is vast and blue. It's more than 20,000 feet deep. There are many fish in it.\",\n",
+    "        Person(name=None, height_in_meters=None, hair_color=None),\n",
```

Comment:
In this case the negative sample seems to hinge on all the attributes of `Person` being optional. Is it worth including another case-- maybe with `Data`, which appears unused in this notebook?

Reply:
i think data should be used down below